### PR TITLE
Update the JS pipeline

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
## Highlights 

🚀 Babel is on the latest version, modern features like `boo ?? foo` and `boo?.name` can be used.
👮‍♂️ Eslint is on the latest version, and new warnings are now available to enforce best practices.
🔧 Format command has been added via Eslint `--fix`. 
🔧 New Github job added to check for lining and dev build posting.
🔥 DevTools now available via `inline-source-map`
🔥 Webpack is on the latest version.
🖌️ SCSS lang now available
🔥 E2E Testing with Playwright for _Free Version_

## Breakdown

The current state of the JS packages is unacceptable, especially the Babel version, which was very old, and many JS features needed to have the required support.

Upgrading the Babel version was a straight process; we removed some old packages and added new ones like `core-js`, which can provide polyfills for old browsers if needed.

Vue was updated to its latest version of supported `2.X`. Related packages like `vuex` and `vue-loader` were also updated to the version that still supports `Vue 2.X` -- in the future, it is recommended to migrate to Vue 3 and Pina store (since it is a new officially recommended store). SASS lang can now be used.

After much refactoring to Webpack, the new version of Webpack has excellent features like `--mode`, which can easily control the build flow without the need for an env and additional lib like `cross-env`. 

Also, the build result is now simplified. The old one created two types of files `build.js` and `build.min.js`, which were coordinated by a PHP var named `ROP_DEBUG`.  Now, only `build.js` is created and loaded, and its content is determined by Webpack `--mode`. 

ℹ️ On the published version of ROP on WordPress, you will see that both `build.js` and `build.min.js` were uploaded. Also, the build files were submitted via Git -- now the `assets/js/build` is excluded via `.gitignore` and the deploy job has been updated to run an `npm run build` before uploading the the version to WordPress.

DevTools are now in full force, `inline-source-map` is now available. Fixing bugs should now be a lot more easier.

The E2E setup is similar to the one from Gutenberg which allows us to use their utility function to deal with WordPress internals.

## Testing

This is a pretty big update and we need to make sure that we did not introduce any regression. Please check each feature in the interface to ensure the components works as expected.

### CODE REVIEW

File: artifacts/storage-states/admin.json
Comment (preda-bogdan) [inline artifacts/storage-states/admin.json:1]: This should not be under `artifacts`.
Why is this required?

File: includes/admin/class-rop-admin.php [hunk 1]
```diff
@@ -300,8 +300,8 @@ public function enqueue_scripts() {
 			return;
 		}
 		wp_enqueue_media();
-		wp_register_script( $this->plugin_name . '-dashboard', ROP_LITE_URL . 'assets/js/build/dashboard' . ( ( ROP_DEBUG ) ? '' : '.min' ) . '.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
-		wp_register_script( $this->plugin_name . '-exclude', ROP_LITE_URL . 'assets/js/build/exclude' . ( ( ROP_DEBUG ) ? '' : '.min' ) . '.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
+		wp_register_script( $this->plugin_name . '-dashboard', ROP_LITE_URL . 'assets/js/build/dashboard.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
+		wp_register_script( $this->plugin_name . '-exclude', ROP_LITE_URL . 'assets/js/build/exclude.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
```
Comment (preda-bogdan) [inline includes/admin/class-rop-admin.php:304]: Instead of `time()` we can use the `ROP_LITE_VERSION` constant, `time()` will bust the cache each time the page is loaded.

File: includes/admin/class-rop-admin.php [hunk 2]
```diff
@@ -374,7 +374,7 @@ public function enqueue_scripts() {
 
 		if ( 'publish_now' === $page ) {
 			$array_nonce['publish_now'] = apply_filters( 'rop_publish_now_attributes', $array_nonce['publish_now'] );
-			wp_register_script( $this->plugin_name . '-publish_now', ROP_LITE_URL . 'assets/js/build/publish_now' . ( ( ROP_DEBUG ) ? '' : '.min' ) . '.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
+			wp_register_script( $this->plugin_name . '-publish_now', ROP_LITE_URL . 'assets/js/build/publish_now.js', array(), ( ROP_DEBUG ) ? time() : $this->version, false );
```
Comment (preda-bogdan) [inline includes/admin/class-rop-admin.php:377]: same here ☝🏼

File: vue/src/models/rop_store.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ Vue.use(Vuex);
 Vue.use(VueResource);
 Vue.use(VueLogger, logOptions);
 
+
+// In the futures, this should be replaced with Pina library.
```
Comment (preda-bogdan) [inline vue/src/models/rop_store.js:32]: we can create an issue for this, else there is a high chance that this will be forgotten.

File: vue/src/vue-elements/account-schedule.vue [hunk 4]
```diff
@@ -1,89 +1,146 @@
...
+        <p class="text-gray">
+          {{ labels.schedule_fixed_time_desc }}
+        </p>
+      </div>
+      <div class="column col-6 col-sm-12 vertical-align">
+        <div class="form-group">
+          <div
+            v-for="( time, index ) in schedule.interval_f.time"
+            class="input-group"
+          >
```
Comment (preda-bogdan) [inline vue/src/vue-elements/account-schedule.vue:69]: A `:key` is missing from the loop, this should be used as to allow better re-render, else caching and selective re-render will not work and excesive re-render will be triggered.

File: vue/src/vue-elements/accounts-selector-panel.vue [hunk 5]
```diff
@@ -1,58 +1,104 @@
...
+      </div>
+      <empty-active-accounts v-if="accountsCount === 0" />
+      <div
+        v-if="accountsCount > 0"
+        class="container"
+      >
+        <div class="columns">
+          <div class="column col-3 col-sm-12 col-md-12 col-xl-3 col-lg-3 col-xs-12 col-rop-selector-accounts">
+            <span class="divider" />
+            <div v-for="( account, id ) in active_accounts">
```
Comment (preda-bogdan) [inline vue/src/vue-elements/accounts-selector-panel.vue:17]: The loop should use a `:key`.

File: vue/src/vue-elements/accounts-selector-panel.vue [hunk 6]
```diff
@@ -141,6 +189,10 @@
 				//this.get_lang();
 			}
 		},
+		mounted: function () {
+			this.setupData();
+		//	this.refresh_language_taxonomies();
```
Comment (preda-bogdan) [inline vue/src/vue-elements/accounts-selector-panel.vue:194]: Is this required or should be removed?

File: vue/src/vue-elements/accounts-tab-panel.vue [hunk 7]
```diff
@@ -1,55 +1,99 @@
...
-                            <add-account-tile></add-account-tile>
-                            <span class="divider"></span>
-                        </div>
-                    </div>
-                </div>
+            <div
+              v-for="( account, id ) in accounts"
+              v-if="is_preloading > 0"
+              class="account-container"
+            >
```
Comment (preda-bogdan) [inline vue/src/vue-elements/accounts-tab-panel.vue:38]: You can wrap this and apply the `if` condition on the wrapper as to not check for each loop item. Also the loop is missing a `:key`

**Review by preda-bogdan — CHANGES_REQUESTED** [review]

File: artifacts/storage-states/admin.json
Comment (Soare-Robert-Daniel) [reply artifacts/storage-states/admin.json:1]: Thanks for spotting it. The folder will be excluded since it will be used by E2E to post results/errors.

File: includes/admin/class-rop-admin.php (hunk 1 above)
Comment (Soare-Robert-Daniel) [reply includes/admin/class-rop-admin.php:304]: I think this is the desired effect since the checking with `ROP_DEBUG` is used. For prod, `$this->version` should create problems.

File: vue/src/vue-elements/accounts-selector-panel.vue (hunk 6 above)
Comment (Soare-Robert-Daniel) [reply vue/src/vue-elements/accounts-selector-panel.vue]: I will removed based on this commit https://github.com/Codeinwp/tweet-old-post/commit/ca44ce0d883238006cc0bfa08ae296fdf532fa32

File: includes/admin/class-rop-admin.php (hunk 2 above)
Comment (Soare-Robert-Daniel) [reply includes/admin/class-rop-admin.php:377]: same response as here: https://github.com/Codeinwp/tweet-old-post/pull/897#discussion_r1415497391

File: vue/src/vue-elements/main-page-panel.vue [hunk 8]
```diff
@@ -1,88 +1,182 @@
...
+            href="https://revive.social"
+            target="_blank"
+          ><b>Revive.Social</b></a> <a
+            id="rop_vid_tuts"
+            href="https://www.youtube.com/playlist?list=PLAsG7vAH40Q512C8d_0lBpVZusUQqUxuH"
+            target="_blank"
+          ><span>START HERE <i
+            class="fa fa-play-circle"
+            aria-hidden="true"
+          /></span></a></span>
```
Comment (preda-bogdan) [inline vue/src/vue-elements/main-page-panel.vue:25]: NIT: The formatting here does not look right.

**Review by preda-bogdan — APPROVED** [review]

File: vue/src/vue-elements/main-page-panel.vue (hunk 8 above)
Comment (Soare-Robert-Daniel) [reply vue/src/vue-elements/main-page-panel.vue]: Thanks. Also, I switched to using the Eslint version from `wp-scripts`.

**Review by GrigoreMihai — APPROVED** [review]
Looks great, just a small thing, we can mention the node version that should be used in the contributing file.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested the PR and didn't find any issue 🚀
